### PR TITLE
Making MessageBuffer a slightly more concrete class.

### PR DESCRIPTION
**Pull request by GhostofCookie**

### CODE REVIEW

File: src/message_buffer.cpp [hunk 1]
```diff
@@ -3,24 +3,90 @@
...
+}
+  
+void MessageBuffer::push_back(const std::vector<uint8_t>& data)
+{
+  _buffer.insert(_buffer.end(), data.begin(), data.end());
+}
+
+void MessageBuffer::pop_back(uint16_t len)
+{
+  const auto delta = _buffer.size() - len;
```
Comment (paulej) [inline src/message_buffer.cpp:46]: Should limit the delta to <= size().

File: src/message_buffer.cpp [hunk 2]
```diff
@@ -3,24 +3,90 @@
...
+  _buffer.erase(_buffer.begin() + delta, _buffer.end());
+};
+
+std::vector<uint8_t> MessageBuffer::back(uint16_t len)
+{
+  assert(len <= _buffer.size());
```
Comment (paulej) [inline src/message_buffer.cpp]: While assert helps in debugging, we should constrain the release builds so they do not return more than the size() of octets.

File: src/message_buffer.cpp [hunk 3]
```diff
@@ -3,24 +3,90 @@
 #include <iomanip>
 #include <sstream>
 
-namespace quicr::messages {
+namespace quicr {
+uintVar_t
+to_varint(uint64_t v)
```
Comment (paulej) [inline src/message_buffer.cpp:8]: This is just an assignment to int.  Do we want a "real" variable length integer?  QUIC defines one for unsigned int.  Cullen defined a different one for GameState, and GSE has both signed and unsigned.

**Review by paulej — DISMISSED** [review]

File: src/message_buffer.cpp (hunk 1 above)
Comment (GhostofCookie) [reply src/message_buffer.cpp:46]: Good catch!

**Review by paulej — APPROVED** [review]

**Pull request merged**